maps/RandomZLevels/tomb.dm

```diff
@@ -260,3 +263,63 @@
 		AM.ex_act(1)
 
 	explosion(T, -1, -1, 1)
+
+
+//SPECIAL BUTTONS
+//Only two can be activated at once
+//Activating a third button when two are activated toggles the first one off
+/obj/structure/button/water_puzzle
```

Comment:
The type name is not particularly descriptive and these might be used for more than just the water puzzle in the future, so I suggest you rename them.